### PR TITLE
chore: onboarding doc updates after test-drive

### PR DESCRIPTION
**Pull request by rb-determined-ai**

### CODE REVIEW

File: onboarding/mlsys/1-gradients.md [hunk 1]
```diff
@@ -491,3 +512,14 @@ reduces network communications.  If you always aggregate gradients from 4
...
+## Summary
+
+Upon completion of this exercise, you have learned:
+
+- the hairy details of gradient descent
+- that averaging gradients allows for faster training
+- how batching data is just a form of averaging gradients
+- how data-parallel dtrain is just a different form of averaging gradients
+- why gradient aggregation (a third form of averaging gradients) is
+  complementary to data-parallel dtrain
```
Comment (tayritenour) [inline onboarding/mlsys/1-gradients.md:525]: Since we capitalize most everything else, we should probably also capitalize these.

File: onboarding/mlsys/1-gradients.md [hunk 2]
```diff
@@ -491,3 +512,14 @@ reduces network communications.  If you always aggregate gradients from 4
 batches before applying them to the model, you only need to communicate
 gradients between workers 1/4th of the time, significantly reducing your
 communication overhead.
+
+## Summary
+
+Upon completion of this exercise, you have learned:
+
+- the hairy details of gradient descent
```
Comment (tayritenour) [inline onboarding/mlsys/1-gradients.md:520]: I would prefer something more neutral like "How to implement gradient descent"

File: onboarding/mlsys/1-gradients.md [hunk 3]
```diff
@@ -210,7 +229,9 @@ Additionally, in the "select one or more `(x, ytrue)` input pairs" step, for
 now you should just select the next `(x, ytrue)` pair in the dataset, and it
 should wrap back to the beginning of the dataset after reaching the end.  So if
 your dataset is of length 10, then your eleventh iteration should reuse the
-first `(x, ytrue)` pair of your dataset.
+first `(x, ytrue)` pair of your dataset.  Effectively this will let you loop
+seamlessly through multiple epochs your dataset, a strategy that is common in
+large-scale machine learning training.
```
Comment (tayritenour) [inline onboarding/mlsys/1-gradients.md:234]: This seems like a fine thing to add but it isn't really what I intended. What I was more going for was an explanation of "why aren't we just using all the data each iteration?" since there's only like 4 datapoints in our examples

**Review by tayritenour — APPROVED** [review]
Some nits

File: onboarding/mlsys/1-gradients.md (hunk 3 above)
Comment (rb-determined-ai) [reply onboarding/mlsys/1-gradients.md:237]: Got it.  That makes sense, especially after the `select one or more` phrasing in the algorithm description.
